doc/design/manifest.md

```diff
@@ -0,0 +1,589 @@
+
+
+## Manifests
+
+Each manifest is a [canonical json](http://wiki.laptop.org/go/Canonical_JSON) file with a `.json` extension.
+
+Manifest format:
+
+```
+{
+    "signatures": [
+        {
+            "keyid": "",
+            "sig": "",
+        }
+    ],
+    "signed": {
+        "_type": "TY",
+        "spec_version": "",
+        "expires": "TS",
+        // version is only present for versioned manifests.
+        "version": n,
+        ROLE
+    },
+}
+```
+
+where
+
+* `""` is a string.
+* `n` is an integer > 0.
+* `"TS"` is a timestamp (date and time) in ISO 8601 format; timezone should always be `Z`.
+* `"TY"` is one of `"root"`, `"index"`, `"component"`, `"snapshot"`, or `"timestamp"`.
+* `ROLE` is specific to the type of manifest.
+* The value of `"spec_version"` should be valid semver.
+
+The timestamp and snapshot manifests should expire after one day. All other manifests should expire after one year.
+
+URLs are relative to the repository root, which allows a repository to be cloned to a new location without changing URLs.
+
+The spec version should be the same for all files in a given snapshot, if the client finds a spec version which is inconsistent, or that the client cannot interpret, it should abort its operation.
+
+### Keys
+
+A key definition is:
+
+```json
+"KEYID": {
+    "keyid_hash_algorithms": [
+        "sha256",
+        "sha512",
+    ],
+    "keytype": "ed25519",
+    "keyval": {
+        "public": "KEY",
+    },
+    "scheme": "ed25519",
+},
+```
+
+where
+
+* `"KEYID"` is a globally unique identifier (i.e., keys must not share key ids); it is the hexdigest of the SHA-256 hash of the canonical JSON form of the key.
+* We support `"sha256"` and `"sha512"` hashing (only SHA256 in first iteration).
+* We support `"rsa"`, `"ed25519"`, and `"ecdsa-sha2-nistp256"` key types (only `"rsa"` in first iteration).
+* We support `"rsassa-pss-sha256"`, `"ed25519"`, and `"ecdsa-sha2-nistp256"` schemes (only `"rsassa-pss-sha256"` in first iteration)
+  - "rsassa-pss-sha256" : RSA Probabilistic signature scheme with appendix. The underlying hash function is SHA256. https://tools.ietf.org/html/rfc3447#page-29
+  - "ed25519" : Elliptic curve digital signature algorithm based on Twisted Edwards curves. https://ed25519.cr.yp.to/
+  - "ecdsa-sha2-nistp256" : Elliptic Curve Digital Signature Algorithm with NIST P-256 curve signing and SHA-256 hashing. https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm
```

Comment:
There's no need to claim to support all these options, unless perhaps we're developing this as a general-purpose library for others to use.

Reply:
I wasn't sure why TUF chose to support these three or their relative merits, so I erred on the side of caution. Supporting just one seems like a good idea, but I didn't know which one to choose! Even if we only choose one, I think it is good to have the flexibility to support others in case we need it in the future (it is not very much extra complexity).